docs/Course Wiki/ECE Course Offerings/ECE391.md

```diff
@@ -0,0 +1,67 @@
+
+## Course Structure
+
+The coursework is entirely MP and homwork based. There are 4 MPs, and MP1 and MP2 have an associated Problem set, each worth 10% of the full MP grade. The problem sets consist of questions relevant to your MPs to help prime you for those assignments.
+
+There are 2 midterms and a final exam for the course, all weighted equally.
+
+The class begins with MP0, which is just helps you setup your ECE391 environment and introduce you to useful concepts in git.
+
+In MP1, you will write x86 Assembly code to perform functions for a simple program. This involves writing a tasklet and i/o control functions, which mainly involve manipulating a data structure with assembly  code. You will also write code to display to the screen in particular ways. 
+
+In MP2, you will implement much of the software for a game with visual graphics. This involves using the ModeX VGA controller to display values on a screen, learning the ModeX protocol to color certain pixels. You will implement parts of the logic for the game itself, as well as specific ways to draw things on screen. A big part of this MP is writing the Tux controller, a driver for an external device with buttons and LEDs to receive inputs and drive outputs. 
+
+In MP3, you will work in a group of 4 to create an operating system from scratch. This operating system will use paging, a virtual memory system. It will also implement a file system, and also multiple device drivers for things like a keyboard and real time clock. The most important part of this MP is implementing system calls, to allow user code to be executed and interact with the kernel, and using your device drivers for input and output. Finally, students implement scheduling to allow for multiple processes to run concurrently.
+
+## Instructors
+
+This course is routinely taught by two professors. Professor Zbigniew Kalbarczyk has taught every recent semester, as well as Professor Steve Lumetta and Professor Nikita Borisov in other semesters.
+
+The course has professors handling the lecture component and the TAs handling the (optional) lab component. Undergraduate course assistants aid with office hours and grading.
+
+## Course Tips
+
+Start on the MPs as soon as you get them. In MP3 Particularly, each new checkpoint builds on the last, so missing parts on one checkpoint can hugely hinder you if you get behind. It is also very valuable to use the lab (ECEB 3026) to work, to get tips from course assistants during office hours, and other students as you work.
```

Comment:
This is known as a stupidly hard course. Is that true? How many hours per week? What can you do to avoid hours and hours of debugging? How to work well together as a group?